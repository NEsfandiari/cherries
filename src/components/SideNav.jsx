import React, { Component } from 'react'
import styled from 'styled-components'
import { NavLink, SideNavLink } from './atoms'
import { StaticQuery, graphql } from 'gatsby'

const Container = styled.div`
  position: fixed;
  margin-top: 3rem;
  z-index: 1;
  overflow: auto;
  height: 70vh;
  #side-navbar-list {
    list-style: none;
  }
  .side-navbar-link {
    font-size: 0.7rem;
  }
  #side-navbar-list li.side-navbar-link,
  a,
  a:visited {
    margin-bottom: 20px;
    text-decoration: none;
    font-weight: 900;
    font-size: 1rem;
    color: #e20031;
  }
  .hyphen {
    margin-left: 7px;
  }
`
class SideNav extends Component {
  render() {
    const links = this.props.links.map((link, i) => {
      if (link.name === '-') {
        return (
          <li key={link.name} className="hyphen side-navbar-link">
            {link.name}
          </li>
        )
      }
      if (!link.dropDown) {
        return (
          <li className="side-navbar-link" key={link.name}>
            <NavLink key={i} to={link.route}>
              {link.name}
            </NavLink>
          </li>
        )
      } else {
        return (
          <SideNavLink
            key={link.name}
            name={link.name}
            dropDown={link.dropDown}
          />
        )
      }
    })
    return (
      <Container>
        <ul id="side-navbar-list">{links}</ul>
      </Container>
    )
  }
}

export default () => (
  <StaticQuery
    query={graphql`
      {
        contentfulHomePage(pageTitle: { eq: "Home Page" }) {
          sideNavBar {
            data {
              name
              displayed
              route
              dropDown {
                name
                route
              }
            }
          }
        }
      }
    `}
    render={data => <SideNav links={data.contentfulHomePage.sideNavBar.data} />}
  />
)
