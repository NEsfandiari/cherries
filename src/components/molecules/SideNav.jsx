import React, { Component } from 'react'
import styled from 'styled-components'
import ShoppingBagIcon from '../atoms/ShoppingBagIcon'
import { Link } from 'gatsby'
import NavLink from '../atoms/NavLink'
import DropdownMenu from '../atoms/DropdownMenu'

/** Overview of NavButtons component:
 *  -
 */

const Container = styled.div`
  .dropdownContent {
    display: none;
  }
  .dropdownContent > * {
    margin-left: 25px;
  }
  ul {
    list-style: none;
  }
  .hyphen {
    margin-left: 7px;
  }
`
//Use the way constructor and normal bind
class SideNav extends Component {
  constructor(props) {
    super(props)
    this.state = {
      display: false,
    }
    this.toggleMenu = this.toggleMenu.bind(this)
  }

  toggleMenu(evt) {
    evt.preventDefault()
    this.setState(st => ({
      display: !st.display,
    }))
  }

  render() {
    const display = this.state.display ? 'initial' : 'none'
    const links = this.props.links.map((link, i) => {
      if (link.name === '-') {
        return (
          <li
            style={{ textDecoration: 'none', color: '#e20031' }}
            className="hyphen"
          >
            {link.name}
          </li>
        )
      }
      if (!link.dropDown) {
        return (
          <li>
            <NavLink
              style={{ textDecoration: 'none', color: '#e20031' }}
              key={i}
              to={link.route}
            >
              {link.name}
            </NavLink>
          </li>
        )
      } else {
        return (
          <li>
            <NavLink
              style={{ textDecoration: 'none', color: '#e20031' }}
              key={i}
              onClick={this.toggleMenu}
            >
              {link.name} +
            </NavLink>
            <ul className="dropdownContent" style={{ display: display }}>
              {link.dropDown.map((li, i) => {
                return (
                  <li>
                    <NavLink
                      style={{ textDecoration: 'none', color: '#47525E' }}
                      key={i}
                      to={li.route}
                    >
                      {li.name}
                    </NavLink>
                  </li>
                )
              })}
            </ul>
          </li>
        )
      }
    })
    return (
      <Container>
        <ul>{links}</ul>
      </Container>
    )
  }
}

export default SideNav
