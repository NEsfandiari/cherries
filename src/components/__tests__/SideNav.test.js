import React from 'react'
import renderer from 'react-test-renderer'
import { SideNav, PureSideNav } from '../SideNav'
describe('SideNav', () => {
  it('renders correctly', () => {
    const data = {
      contentfulHomePage: {
        sideNavBar: {
          data: [
            {
              name: 'Vibes',
              displayed: 'either',
              route: null,
              dropDown: [
                {
                  name: 'Basics',
                  route: 'basics',
                },
                {
                  name: 'Minimalist',
                  route: 'minimalist',
                },
                {
                  name: 'Girlboss',
                  route: 'girlboss',
                },
                {
                  name: 'Formal',
                  route: 'formal',
                },
                {
                  name: 'Boho',
                  route: 'boho',
                },
                {
                  name: 'Fun',
                  route: 'fun',
                },
              ],
            },
            {
              name: 'Type',
              displayed: 'either',
              route: null,
              dropDown: [
                {
                  name: 'Stud',
                  route: 'stud',
                },
                {
                  name: 'Hoop',
                  route: 'hoop',
                },
                {
                  name: 'Drop',
                  route: 'drop',
                },
                {
                  name: 'Dangle',
                  route: 'dangle',
                },
                {
                  name: 'Huggy',
                  route: 'huggy',
                },
                {
                  name: 'Thread',
                  route: 'thread',
                },
              ],
            },
            {
              name: 'Finish',
              displayed: 'either',
              route: null,
              dropDown: [
                {
                  name: 'Resin',
                  route: 'resin',
                },
                {
                  name: 'Enamel',
                  route: 'enamel',
                },
                {
                  name: 'Gold',
                  route: 'gold',
                },
                {
                  name: 'Silver',
                  route: 'silver',
                },
              ],
            },
            {
              name: 'Shop All',
              displayed: 'either',
              route: 'catalog',
              dropDown: null,
            },
            {
              name: '-',
              displayed: 'either',
              route: null,
              dropDown: null,
            },
            {
              name: 'Subscribe',
              displayed: 'userNotSubscribed',
              route: 'subscribe',
              dropDown: null,
            },
            {
              name: 'About',
              displayed: 'either',
              route: 'about',
              dropDown: null,
            },
          ],
        },
      },
    }
    const tree = renderer.create(<PureSideNav data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
