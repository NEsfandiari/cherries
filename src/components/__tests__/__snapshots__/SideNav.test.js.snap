// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav renders correctly 1`] = `
<div
  className="sc-jTzLTM gwmvbb"
>
  <ul
    id="side-navbar-list"
  >
    <div
      className="sc-EHOje ciWOoP"
    >
      <li
        className="side-navbar-link"
      >
        <a
          className="sc-bxivhb dToeqL"
          onClick={[Function]}
        >
          Vibes
           
          +
        </a>
        <ul
          className="side-navbar-dropdown"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <li
            className="side-navbar-dropdown-link"
          >
            <a
              className="sc-bxivhb dToeqL"
              href="basics"
            >
              Basics
            </a>
          </li>
          <li
            className="side-navbar-dropdown-link"
          >
            <a
              className="sc-bxivhb dToeqL"
              href="minimalist"
            >
              Minimalist
            </a>
          </li>
          <li
            className="side-navbar-dropdown-link"
          >
            <a
              className="sc-bxivhb dToeqL"
              href="girlboss"
            >
              Girlboss
            </a>
          </li>
          <li
            className="side-navbar-dropdown-link"
          >
            <a
              className="sc-bxivhb dToeqL"
              href="formal"
            >
              Formal
            </a>
          </li>
          <li
            className="side-navbar-dropdown-link"
          >
            <a
              className="sc-bxivhb dToeqL"
              href="boho"
            >
              Boho
            </a>
          </li>
          <li
            className="side-navbar-dropdown-link"
          >
            <a
              className="sc-bxivhb dToeqL"
              href="fun"
            >
              Fun
            </a>
          </li>
        </ul>
      </li>
    </div>
    <div
      className="sc-EHOje ciWOoP"
    >
      <li
        className="side-navbar-link"
      >
        <a
          className="sc-bxivhb dToeqL"
          onClick={[Function]}
        >
          Type
           
          +
        </a>
        <ul
          className="side-navbar-dropdown"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <li
            className="side-navbar-dropdown-link"
          >
            <a
              className="sc-bxivhb dToeqL"
              href="stud"
            >
              Stud
            </a>
          </li>
          <li
            className="side-navbar-dropdown-link"
          >
            <a
              className="sc-bxivhb dToeqL"
              href="hoop"
            >
              Hoop
            </a>
          </li>
          <li
            className="side-navbar-dropdown-link"
          >
            <a
              className="sc-bxivhb dToeqL"
              href="drop"
            >
              Drop
            </a>
          </li>
          <li
            className="side-navbar-dropdown-link"
          >
            <a
              className="sc-bxivhb dToeqL"
              href="dangle"
            >
              Dangle
            </a>
          </li>
          <li
            className="side-navbar-dropdown-link"
          >
            <a
              className="sc-bxivhb dToeqL"
              href="huggy"
            >
              Huggy
            </a>
          </li>
          <li
            className="side-navbar-dropdown-link"
          >
            <a
              className="sc-bxivhb dToeqL"
              href="thread"
            >
              Thread
            </a>
          </li>
        </ul>
      </li>
    </div>
    <div
      className="sc-EHOje ciWOoP"
    >
      <li
        className="side-navbar-link"
      >
        <a
          className="sc-bxivhb dToeqL"
          onClick={[Function]}
        >
          Finish
           
          +
        </a>
        <ul
          className="side-navbar-dropdown"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <li
            className="side-navbar-dropdown-link"
          >
            <a
              className="sc-bxivhb dToeqL"
              href="resin"
            >
              Resin
            </a>
          </li>
          <li
            className="side-navbar-dropdown-link"
          >
            <a
              className="sc-bxivhb dToeqL"
              href="enamel"
            >
              Enamel
            </a>
          </li>
          <li
            className="side-navbar-dropdown-link"
          >
            <a
              className="sc-bxivhb dToeqL"
              href="gold"
            >
              Gold
            </a>
          </li>
          <li
            className="side-navbar-dropdown-link"
          >
            <a
              className="sc-bxivhb dToeqL"
              href="silver"
            >
              Silver
            </a>
          </li>
        </ul>
      </li>
    </div>
    <li
      className="side-navbar-link"
    >
      <a
        className="sc-bxivhb dToeqL"
        href="catalog"
      >
        Shop All
      </a>
    </li>
    <li
      className="hyphen side-navbar-link"
    >
      -
    </li>
    <li
      className="side-navbar-link"
    >
      <a
        className="sc-bxivhb dToeqL"
        href="subscribe"
      >
        Subscribe
      </a>
    </li>
    <li
      className="side-navbar-link"
    >
      <a
        className="sc-bxivhb dToeqL"
        href="about"
      >
        About
      </a>
    </li>
  </ul>
</div>
`;
